Firmware/Marlin_main.cpp

```diff
@@ -11282,10 +11288,13 @@ void restore_print_from_eeprom(bool mbl_was_active) {
 
     // Move to the XY print position in logical coordinates, where the print has been killed, but
     // without shifting Z along the way. This requires performing the move without mbl.
-	sprintf_P(cmd, PSTR("G1 X%f Y%f F3000"),
-              eeprom_read_float((float*)(EEPROM_UVLO_CURRENT_POSITION + 0)),
-              eeprom_read_float((float*)(EEPROM_UVLO_CURRENT_POSITION + 4)));
-	enquecommand(cmd);
+    float pos_x = eeprom_read_float((float*)(EEPROM_UVLO_CURRENT_POSITION + 0));
+    float pos_y = eeprom_read_float((float*)(EEPROM_UVLO_CURRENT_POSITION + 4));
+    if (pos_x != X_COORD_INVALID)
+    {
+        sprintf_P(cmd, PSTR("G1 X%f Y%f F3000"), pos_x, pos_y);
```

Comment:
are we sure we never run out of the cmd buffer with different float format?

Reply:
You mean something which could be potentially broken?

x/y were always stored during uvlo_ (and during crash detection, which is handled the same just in a different function), so it's highly unlikely that you'll be able to flash the fw just after a power panic, but before the printer resumes (changing the value is not an issue either).

The same method is also used to mark the "saved target" (for multipath G1 commands) as invalid.
This is less annoying that adding a new global+eeprom value to mark the value as invalid IMHO.

I'm currently defining X_COORD_INVALID as X_MIN-1, which is guaranteed to not pass the frontend in marlin (it's going to be clamped to X_MIN-X_MAX).

Reply:
No, I meant if there is a possibility of creating `G1 X<some float> Y<some float> F3000` which runs of those 30 bytes allocated for the cmd buffer. Adding some limitations like `%4.2f` could solve the problem (avoiding the need for snprinf())

Reply:
Got it, I'm going to check that

Reply:
So there's indeed a buffer overflow, which is caused by the subsequent sprintf.
I expanded the buffer to fix the maximum practical limit.

However, sprintf doesn't allow to specify a maximum field width for floats that I know of, it's always a minimum which doesn't help to ensure we're not going to overflow the buffer.

We would need to check all float values manually, which feels expensive code-wise.